packages/engine/src/scene/systems/RendererUpdateSystem.ts

```diff
@@ -10,12 +11,18 @@ import { System } from '../../ecs/classes/System'
  */
 
 const renderedQuery = defineQuery([Object3DComponent, RenderedComponent])
+const renderedUpdatableQuery = defineQuery([UpdatableComponent, RenderedComponent])
 
 export default async function RendererUpdateSystem(world: World): Promise<System> {
   return () => {
     for (const entity of renderedQuery()) {
       const obj = getComponent(entity, Object3DComponent)
       ;(obj.value as unknown as Updatable).update(world.delta)
     }
+
+    for (const entity of renderedUpdatableQuery()) {
```

Comment:
do we need a second query here? If my understanding is correct, Object3DComponent should be the only component that holds the Object3D, the Updateable component a tag component for running an update function on the fixed loop, and RenderedComponent is a tag component for running an update function on the free loop. I think for the interactable UI, we want to only use the RenderedComponent and not use Updateable component at all.

Reply:
Okay got it.